python_modules/libraries/dagstermill/dagstermill/asset_factory.py

```diff
@@ -205,6 +222,7 @@ def iris_dataset():
 
     check.str_param(name, "name")
     check.str_param(notebook_path, "notebook_path")
+    check.opt_bool_param(save_notebook_on_failure, "save_notebook_on_failure")
```

Comment:
can do `save_notebook_on_failure = check.opt_bool_param(save_notebook_on_failure, "save_notebook_on_failure")` so when `save_notebook_on_failure` is not provided by the user, we fill it in as `false`